include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -1620,6 +1622,82 @@ namespace internal
 } // namespace internal
 
 
+
+namespace MGTransferGlobalCoarseningTools
+{
+  template <int dim, int spacedim>
+  std::vector<std::shared_ptr<Triangulation<dim, spacedim>>>
+  create_geometric_coarsening_sequence(
+    const Triangulation<dim, spacedim> &fine_triangulation_in)
+  {
+    std::vector<std::shared_ptr<Triangulation<dim, spacedim>>>
+      coarse_grid_triangulations;
+
+#ifndef DEAL_II_WITH_P4EST
+    (void)fine_triangulation_in;
+    Assert(false, ExcNotImplemented());
+#else
+    const auto fine_triangulation =
+      dynamic_cast<const parallel::distributed::Triangulation<dim, spacedim> *>(
+        &fine_triangulation_in);
+
+    Assert(fine_triangulation, ExcNotImplemented());
+
+    const unsigned int max_level = fine_triangulation->n_global_levels() - 1;
+
+    const auto coarse_mesh_description =
+      GridTools::get_coarse_mesh_description(*fine_triangulation);
+
+    coarse_grid_triangulations.resize(fine_triangulation->n_global_levels());
+
+    coarse_grid_triangulations[max_level].reset(
+      const_cast<Triangulation<dim, spacedim> *>(&fine_triangulation_in),
+      [](auto &) {});
+
+    // create coarse meshes
+    for (unsigned int l = max_level - 1; l != static_cast<unsigned int>(-1);
+         --l)
+      {
+        // store finer mesh to file
+        dynamic_cast<
+          const parallel::distributed::Triangulation<dim, spacedim> *>(
+          coarse_grid_triangulations[l + 1].get())
+          ->save("mesh");
```

Comment:
Are you planning to still work on this (sorry if I missed a discussion on this elsewhere)? It feels unsafe to write to the current directory without the user giving a chance to specify the location. Think of a supercomputer file system where your executable might sit in a place where you don't want to write binary output. I would be much happier if we could write to RAM, but I realize it is not possible with the current infrastructure because we need to synchronize between the MPI ranks. So I guess we will need a somewhat adapted functionality here. Also, it seems we leave a new file around, which is surprising as a user never requested one, do we?

Reply:
Yes, we have plans. See https://github.com/dealii/dealii/pull/11707#pullrequestreview-585109577.

Reply:
@kronbichler Went faster than expected: #11715